enhancements/network/user-defined-network-segmentation.md

```diff
@@ -521,13 +522,24 @@ type UserDefinedNetworkSpec struct {
     // Supported by Topology `Layer2` and `Localnet`.
     // +optional
     IPAMLifecycle NetworkIPAMLifecycle `json:"ipamLifecycle,omitempty"`
+    
+    // Control how much of the IP configuration will be managed by OVN.
+    // When `FullyManaged`, OVN-Kubernetes will apply IP configuration to the SDN infrastructure and it will also assign IPs from the selected subnet to the
+    // individual pods.
+    // When `Disabled`, OVN-Kubernetes will only provide layer 2 communication, letting users configure IP addresses for the pods. `Disabled` is only available
+    // for `Layer2` and `Localnet` topologies. By disabling IPAM, you are also opting-out from features depending on it, e.g. IP spoof filtering and
+    // namespace/pod selectors in network policies.
```

Comment:
it's described here. This is more accurate.

Reply:
We only allow disabling IPAM option on secondary (non-primary) UDNs of types Layer2 and localnet. Currently supported features on secondary networks with ovn-k are
- e-w connectivity
- Port security (IP and MAC spoofing protection)
- multi-network-policy

When IPAM is disabled, the following feature subset won't work:
- IP spoofing protection
- multi-network-policy with namespace/pod selectors

Or what will still work:
- e-w connectivity (if you manage to correctly assign IPs)
- MAC spoofing protection
- multi-network-policies with ipBlock selectors

We have 3 options on documenting this:
1. List what is NOT supported with disabled IPAM. If we ever add a new feature for secondary networks and forget to update this list, it will be "automatically" supported from the CRD comment
2. List what IS supported with disabled IPAM. May be harder to understand the difference with managed IPAM if the user doesn't know the full list of features. On the bright side, any new features for secondary network are not "supported" until we update the comment.
3. Point to some external docs for more information on side-effect and feature support with disabled IPAM.

@trozet any preference on how we want to document this?

Reply:
I would be more generic and say something like this: "By disabling IPAM, any Kubernetes features that rely on selecting pods by IP will no longer function (such as network policy, services, etc). Additionally, IP port security will also be disabled for interfaces attached this network."